### PR TITLE
Code cleanup and linting

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
- Added missing dependencies and configuration values
 - Removed unnecessary duplication of accessing the `Server` object  both `application.server` and passing `server` as argument to certain methods. The Server object is now "protected" on the application object and always passed when needed.
 - Removed unnecessary wrapping of the `mxcubecore` import to access the `beamline` object and using `from mxcubecore import HardwareRepository as HWR` when possible\
 - Linting

### CODE REVIEW

File: test/HardwareObjectsMockup.xml/mxcube-server-config.yml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 server:
   SECRET_KEY: "ASECRETKEY"
+  SECURITY_PASSWORD_SALT = "ASALT"
```
[the remark this answers is not in the document]
Comment (marcus-oscarsson) [reply test/HardwareObjectsMockup.xml/mxcube-server-config.yml]: Thanks :)

**Review by rhfogh — COMMENTED** [review]
- The changes introducing HWR are very welcome  - thanks - I wish I had seen them before I added my last contribution to the discussion. 

- The refactoring around 'server' looks like it is not quite complete, there are both ._server, and ._application._server. BTW, is it true to say that server is passed in in parallel to app in many places, yet is also accessible as app._server? This should maybe be considered, but need not hold up this PR.
More tomorrow.

**Comment by marcus-oscarsson** [comment]
Ah, yes it has taken me a while to publish these changes, there's been quite a lot coming in between. application._server is not really meant to be used by other objects than the Application object itself. I noticed that there is a reference to application._server left outside Application, I will take care of that. I could also make it "private" with __ if you would think that would be better.

I otherwise preferred to pass server as an argument to the various parts that needs it, because its used as a decorator for the routes and the access to it becomes quite clumsy if one needs to write `@application.server.validate` instead of simply `@server.validate . It was in this sense `nicer` to just pass server and I decided to do like that everywhere to be consistent. I would not mind doing the opposite if that what everybody else thinks would be better. Its open for dicussion.

**Review by rhfogh — APPROVED** [review]
Lots of work here. Thanks.

**Pull request merged**